index.html

```diff
@@ -224,36 +224,67 @@
+          canvas if the <a>origin-clean</a> flag of the canvas becomes false.
         </p>
         <p>
-          A <a>user agent</a> SHOULD <a
-          href="http://www.w3.org/TR/html5/webappapis.html#await-a-stable-state">await
-          a stable state</a> in the script execution of the current page or
-          worker that has control of the canvas before capturing a frame.
+          Each track that captures a canvas has
+          a <code><dfn>frameCaptureRequested</dfn></code> property that is set to
+          true when a new frame is requested from the canvas.
+        </p>
+        <p>
+          The value of the <code><a>frameCaptureRequested</a></code> property on
+          all new tracks is set to <code>true</code> when the track is created.
+          On creation of the captured track with a specific,
+          non-zero <code>frameRate</code>, the <a>user agent</a> starts a
+          periodic timer at an interval of <code>1/frameRate</code> seconds.  At
+          each activation of the timer,
+          the <code><a>frameCaptureRequested</a></code> property is set
+          to <code>true</code>.
         </p>
         <p>
           In order to support manual control of frame capture, browsers MUST
-          support a value of 0 for <code>frameRate</code>.  The captured stream
-          always captures at least one frame, even if <code>frameRate</code> is
-          zero.
+          support a value of 0 for <code>frameRate</code>.
         </p>
         <p>
-          If the <code>frameRate</code> value is omitted, the <a>user agent</a>
-          SHOULD capture new frames each time that the content of the canvas
-          changes.
+          A new frame is requested from the canvas
+          when <code><a>frameCaptureRequested</a></code> is true and the canvas is
+          drawn.  Each time that the captured canvas is drawn, the following
+          steps are executed:
+          <ol>
+            <li>
+              For each <var>track</var> capturing from the canvas execute the
+              following steps:
+              <ol>
+                <li>
+                  If the <code><a>frameCaptureRequested</a></code> property
+                  of <var>track</var> is set, add a new frame
+                  to <var>track</var> containing what was drawn to the canvas.
+                </li>
+                <li>
+                  If a <code>frameRate</code> value was specified, set
+                  the <code><a>frameCaptureRequested</a></code> property
+                  of <var>track</var> to <code>false</code>.  As a result,
+                  captures that do not specify a frame rate will capture new
+                  frames each time that the content of the canvas is drawn.
+                  Note that for optimization reasons, a <a>user agent</a> SHOULD
+                  skip capturing a frame if the contents of the canvas have not
+                  changed since the last captured frame.
```

Comment:
Hmm, the condition needs to be on the capture step, otherwise `frameCaptureRequested` never clears.

Also, what do you think we should do for new captures of unchanging canvas elements?